addons/website/static/src/js/content/menu.js

```diff
@@ -814,6 +815,39 @@ publicWidget.registry.MegaMenuDropdown = publicWidget.Widget.extend({
         this.options.wysiwyg?.odooEditor.observerActive("moveMegaMenu");
     },
 
+    /**
+     * @private
+     */
+    _updateActiveMenuLinks() {
+        // Prevent having several active links in the menu.
+        if (this.el.querySelector(".navbar #top_menu a.nav-link.active")) {
+            return;
+        }
+        const currentPathName = window.location.pathname;
+
+        // Check and update the active state of menu items based on the current
+        // page
+        const megaMenuEls = this.el.querySelectorAll(".o_mega_menu");
+        let matchingLink = null;
+        megaMenuEls.forEach((megaMenuEl, position) => {
+            const linkEls = Array.from(megaMenuEl.querySelectorAll(`a:not([href="#"])`));
+            matchingLink = linkEls.find((linkEl) => linkEl.pathname === currentPathName);
+            if (matchingLink) {
+                const megaMenuToggleEl = megaMenuEl
+                    .closest(".nav-item")
+                    .querySelector(".o_mega_menu_toggle");
+                // Target the corresponding link in the mobile navigation. Since the
+                // mega-menu for mobile is dynamically rendered, it is not
+                // accessible at this moment.
+                const mobileMegaMenuToggleEl = this.el.querySelectorAll(
```

Comment:
no need to use `querySelectorAll` here as you're only expecting to match a single element. ;)

Reply:
I had to take the `position` element of the list as commented

```js
const mobileMegaMenuToggleEl = this.el.querySelectorAll(
    "#top_menu_collapse_mobile .top_menu .o_mega_menu_toggle"
)[position];
```

Reply:
@SergeBayet Indeed 👍🏼 so it should be `mobileMegaMenuToggleEls` (with an s)

Reply:
Indeed my bad